drivers/lis2dh12/lis2dh12.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers_lis2dh12
+ * @{
+ *
+ * @file
+ * @brief       LIS2DH12 accelerometer driver implementation
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "assert.h"
+
+#include "lis2dh12.h"
+#include "lis2dh12_internal.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+/* SPI bus speed and mode */
+#define CLK                 SPI_CLK_5MHZ
+#define MODE                SPI_MODE_0
+
+#define BUS_OK              SPI_OK
+
+/* shortcuts for SPI bus parameters */
+#define BUS                 (dev->p->spi)
+#define CS                  (dev->p->cs)
+
+/* flag to set when reading from the device */
+#define READ                (0x80)
+
+/* flag to enable address auto incrementation on read or write */
+#define AUTOINC             (0x40)
+
+static inline int acquire(const lis2dh12_t *dev)
+{
+    return spi_acquire(BUS, CS, MODE, CLK);
+}
+
+static inline void release(const lis2dh12_t *dev)
+{
+    spi_release(BUS);
+}
+
+static inline uint8_t read(const lis2dh12_t *dev, uint8_t reg)
```

Comment:
while this might work, IMO name clashes with standard functions should be avoided. ```_[a-z]``` is guaranteed to be free. ;)